### PR TITLE
refactored pom dependencies for #1012

### PR DESCRIPTION
**Pull request by victoraysev**
refactored pom dependencies

- removed `<version>${project.version}</version>` in every submodule pom
- added a `<dependencyManagement>` block to the central POM file and included the module dependencies with `<version>${project.version}</version>` within it.

opened for the first task of #1012

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -412,7 +412,47 @@
...
+        <dependency>
+            <groupId>org.soot-oss</groupId>
+            <artifactId>sootup.java.core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.soot-oss</groupId>
+            <artifactId>sootup.java.bytecode</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
```
Comment (swissiety) [inline pom.xml]: I think we should not define the scope here as this can be different across the submodules.

**Review by swissiety — COMMENTED** [review]
:+1:

File: pom.xml (hunk 1 above)
Comment (victoraysev) [reply pom.xml]: Thank you for your review @swissiety. I fixed it and commited

**Review by swissiety — APPROVED** [review]
:+1:

**Comment by victoraysev** [comment]
Hi @swissiety, is there anything I can address or improve to help get this pull request accepted?

**Comment by swissiety** [comment]
Hi,
the pipeline checks have to run through. Have a look at the log to see what needs to be fixed - seems it can not compile currently.

**Comment by victoraysev** [comment]
Hi @swissiety 
Thanks for your assistance.
I have added new commits to resolve checks

**Review by swissiety — APPROVED** [review]
thx!

File: sootup.core/pom.xml [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 
 	<name>SootUp Core</name>
-    <artifactId>sootup.core</artifactId>
+	<artifactId>sootup.core</artifactId>
 	<packaging>jar</packaging>
 	<parent>
 		<groupId>org.soot-oss</groupId>
 		<artifactId>sootup</artifactId>
 		<version>1.3.1-SNAPSHOT</version>
+		<relativePath>../pom.xml</relativePath>
```
Comment (swissiety) [inline sootup.core/pom.xml]:
```suggestion
```

see https://www.baeldung.com/maven-relativepath

**Pull request merged**